source/extensions/tracers/zipkin/zipkin_tracer_impl.cc

```diff
@@ -75,10 +75,13 @@ Driver::Driver(const Json::Object& config, Upstream::ClusterManager& cluster_man
   const bool trace_id_128bit =
       config.getBoolean("trace_id_128bit", ZipkinCoreConstants::get().DEFAULT_TRACE_ID_128BIT);
 
-  tls_->set([this, collector_endpoint, &random_generator, trace_id_128bit](
+  const bool shared_span_context = config.getBoolean(
```

Comment:
FWIW, this looks correct but I think we should eventually rip out all the JSON related config stuff in the Zipkin driver and use typed protos; the config is specified in proto and it's more convenient to work at a typed level rather than intermediate dynamically typed JSON configs. This is what happens across the core. This is not actionable in this PR though (maybe file an issue?).

Reply:
Issue already exists: https://github.com/envoyproxy/envoy/issues/3459